File: app/src/main/java/com/example/appdatmon/ui/admin/KitchenAdapter.java
package com.example.appdatmon.ui.admin;

import android.content.res.ColorStateList;
import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.example.appdatmon.R;
import com.example.appdatmon.data.api.RetrofitClient;
import com.example.appdatmon.data.model.Order;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class KitchenAdapter extends RecyclerView.Adapter<KitchenAdapter.KitchenViewHolder> {

    private List<KitchenOrder> orderList;

    public KitchenAdapter(List<KitchenOrder> orderList) {
        this.orderList = orderList;
    }

    public void updateData(List<KitchenOrder> newList) {
        this.orderList = newList;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public KitchenViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_kitchen_order, parent, false);
        return new KitchenViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull KitchenViewHolder holder, int position) {
         KitchenOrder item = orderList.get(position);

        holder.txtTable.setText(item.getTableName());
        holder.txtFood.setText(item.getFoodName());
        holder.txtQuantity.setText("Số lượng: " + item.getQuantity());
        
        String statusLabel = "Chờ chế biến";
        if (item.getStatus().equals("PREPARING")) statusLabel = "Đang chế biến";
        else if (item.getStatus().equals("READY")) statusLabel = "Chờ phục vụ";
        holder.txtStatus.setText(statusLabel);

        updateUIByStatus(holder, item.getStatus());

        holder.btnUpdate.setOnClickListener(v -> {
            String nextStatus = "";
            if (item.getStatus().equals("CONFIRMED")) nextStatus = "PREPARING";
            else if (item.getStatus().equals("PREPARING")) nextStatus = "READY";

            if (!nextStatus.isEmpty()) {
                updateOrderStatusOnServer(item.getId(), nextStatus, position, holder.itemView);
            }
        });
    }

    private void updateOrderStatusOnServer(long orderId, String status, int position, View view) {
        Map<String, String> body = new HashMap<>();
        body.put("status", status);

        RetrofitClient.getOrderApi().updateOrderStatus(orderId, body).enqueue(new Callback<Order>() {
            @Override
            public void onResponse(Call<Order> call, Response<Order> response) {
                if (response.isSuccessful()) {
                    Toast.makeText(view.getContext(), "Cập nhật thành công", Toast.LENGTH_SHORT).show();
                    // Local update
                    orderList.get(position).setStatus(status);
                    notifyItemChanged(position);
                }
            }

            @Override
            public void onFailure(Call<Order> call, Throwable t) {
                Toast.makeText(view.getContext(), "Lỗi kết nối", Toast.LENGTH_SHORT).show();
            }
        });
    }

    private void updateUIByStatus(KitchenViewHolder holder, String status) {
        switch (status) {
            case "CONFIRMED":
                holder.txtStatus.setTextColor(Color.parseColor("#FF9800"));
                holder.txtStatus.setBackgroundTintList(ColorStateList.valueOf(Color.parseColor("#FFF3E0")));
                holder.btnUpdate.setText("BẮT ĐẦU CHẾ BIẾN");
                holder.btnUpdate.setBackgroundTintList(ColorStateList.valueOf(Color.parseColor("#4CAF50")));
                holder.btnUpdate.setVisibility(View.VISIBLE);
                break;
            case "PREPARING":
                holder.txtStatus.setTextColor(Color.parseColor("#2196F3"));
                holder.txtStatus.setBackgroundTintList(ColorStateList.valueOf(Color.parseColor("#E3F2FD")));
                holder.btnUpdate.setText("CHẾ BIẾN XONG");
                holder.btnUpdate.setBackgroundTintList(ColorStateList.valueOf(Color.parseColor("#FF5722")));
                holder.btnUpdate.setVisibility(View.VISIBLE);
                break;
            case "READY":
                holder.txtStatus.setTextColor(Color.parseColor("#4CAF50"));
                holder.txtStatus.setBackgroundTintList(ColorStateList.valueOf(Color.parseColor("#E8F5E9")));
                holder.btnUpdate.setVisibility(View.GONE);
                break;
            default:
                holder.btnUpdate.setVisibility(View.GONE);
                break;
        }
    }

    @Override
    public int getItemCount() {
        return orderList.size();
    }

    static class KitchenViewHolder extends RecyclerView.ViewHolder {
        TextView txtTable, txtFood, txtQuantity, txtStatus;
        Button btnUpdate;

        public KitchenViewHolder(@NonNull View itemView) {
            super(itemView);
            txtTable = itemView.findViewById(R.id.txtTable);
            txtFood = itemView.findViewById(R.id.txtFood);
            txtQuantity = itemView.findViewById(R.id.txtQuantity);
            txtStatus = itemView.findViewById(R.id.txtStatus);
            btnUpdate = itemView.findViewById(R.id.btnUpdate);
        }
    }
}

File: app/src/main/java/com/example/appdatmon/ui/kitchen/KitchenAdapter.kt
package com.example.appdatmon.ui.kitchen

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.appdatmon.R
import com.example.appdatmon.data.model.KitchenItem
import com.google.android.material.button.MaterialButton

class KitchenAdapter(
    private var itemList: List<KitchenItem>,
    private val onActionClick: (KitchenItem, String) -> Unit
) : RecyclerView.Adapter<KitchenAdapter.KitchenViewHolder>() {

    class KitchenViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvTableNumber: TextView = itemView.findViewById(R.id.tvTableNumber)
        val tvOrderTime: TextView = itemView.findViewById(R.id.tvOrderTime)
        val tvItemStatus: TextView = itemView.findViewById(R.id.tvItemStatus)
        val ivDishImage: ImageView = itemView.findViewById(R.id.ivDishImage)
        val tvDishName: TextView = itemView.findViewById(R.id.tvDishName)
        val tvDishQuantity: TextView = itemView.findViewById(R.id.tvDishQuantity)
        val layoutNote: LinearLayout = itemView.findViewById(R.id.layoutNote)
        val tvItemNote: TextView = itemView.findViewById(R.id.tvItemNote)
        val btnAction: MaterialButton = itemView.findViewById(R.id.btnAction)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): KitchenViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_kitchen_order, parent, false)
        return KitchenViewHolder(view)
    }

    override fun onBindViewHolder(holder: KitchenViewHolder, position: Int) {
        val item = itemList[position]

        val tableNumber = item.Order?.RestaurantTable?.tableNumber
        holder.tvTableNumber.text = if (tableNumber != null) "Bàn số $tableNumber" else "Mang đi"
        holder.tvOrderTime.text = "(Đơn #${item.Order?.id ?: "N/A"})"
        
        holder.tvDishName.text = item.Product?.name ?: "Món ăn"
        holder.tvDishQuantity.text = "Số lượng: ${item.quantity}"

        // Handle Image
        val imageUrl = item.Product?.image
        if (!imageUrl.isNullOrEmpty()) {
            val fullUrl = "http://10.0.2.2:3000$imageUrl"
            Glide.with(holder.itemView.context)
                .load(fullUrl)
                .placeholder(android.R.drawable.ic_menu_gallery)
                .into(holder.ivDishImage)
        } else {
            holder.ivDishImage.setImageResource(android.R.drawable.ic_menu_gallery)
        }

        // Handle Note
        if (!item.note.isNullOrEmpty()) {
            holder.layoutNote.visibility = View.VISIBLE
            holder.tvItemNote.text = item.note
        } else {
            holder.layoutNote.visibility = View.GONE
        }

        // Handle status badge and action button based on current status
        when (item.status.uppercase()) {
            "WAITING" -> {
                holder.tvItemStatus.text = "Chờ chế biến"
                holder.tvItemStatus.setBackgroundResource(R.drawable.bg_status_waiting)
                
                holder.btnAction.text = "Bắt đầu nấu"
                holder.btnAction.setBackgroundColor(ContextCompat.getColor(holder.itemView.context, android.R.color.holo_orange_dark))
                
                holder.btnAction.setOnClickListener {
                    onActionClick(item, "COOKING")
                }
            }
            "COOKING" -> {
                holder.tvItemStatus.text = "Đang chế biến"
                holder.tvItemStatus.setBackgroundResource(R.drawable.bg_status_cooking)
                
                holder.btnAction.text = "Hoàn thành"
                holder.btnAction.setBackgroundColor(ContextCompat.getColor(holder.itemView.context, android.R.color.holo_green_dark))
                
                holder.btnAction.setOnClickListener {
                    onActionClick(item, "DONE")
                }
            }
            else -> {
                holder.tvItemStatus.text = item.status
                holder.tvItemStatus.setBackgroundResource(R.drawable.bg_status_inactive)
                holder.btnAction.visibility = View.GONE
            }
        }
    }

    override fun getItemCount(): Int = itemList.size

    fun updateData(newList: List<KitchenItem>) {
        itemList = newList
        notifyDataSetChanged()
    }
}
